src/student/components/quizView/QuestionsList.tsx

```diff
@@ -10,20 +10,48 @@ export function QuestionsList() {
     let answers = new Map();
     const [quiz, setQuiz] = useState(testData());
     const [quizID, setQuizID] = useState("");
+
+    const fillcheckboxValues = () => {
+        let values : boolean[][] = [];
+        for(let i = 0; i < quiz.questions.length; i++) {
+            values[i] = []
+            let len;
+            if(quiz.questions[i].options === undefined) continue;
+            else{  len = quiz.questions[i].options!.length;}
+            for(let j = 0; j < len; j++)
+                values[i].push(false);
+            
+        }
+        return values
+    }
+
+    const [checkboxValues, setCheckboxValues] = useState<boolean[][]>(fillcheckboxValues());
     const store = useContext(StoreContext);
     const { socketEmiter, sendJsonMessage } = useSocket();
+    
     const classes = makeStyles({
         details: {
             padding: "20px 10px",
             background: "#fedf9d",
             overflow: 'auto',
         }
     })();
-    // 
+
     const refreshQuiz = useCallback((payload: ServerQuizRequestPayload) => {
         console.log("refreshQuiz");
         setQuiz(payload.data.questions);
         setQuizID(payload.data.quiz_id);
+        let values : boolean[][] = [];
+        for(let i = 0; i < quiz.questions.length; i++) {
+            values[i] = []
+            let len;
+            if(quiz.questions[i].options === undefined) continue;
+            else{  len = quiz.questions[i].options!.length;}
+            for(let j = 0; j < len; j++)
+                values[i].push(false);
+            
+        }
+        setCheckboxValues(values);
     }, []);
```

Comment:
You are setting new quiz data but setQuiz only schedules the change of state and it will be done in the next rerender of the component. You have to use the quiz from the payload in the loop and not quiz the state because it hasn't updated yet. Maybe change the fillCheckboxValues function to accept the quiz object on input and use that?
```suggestion
    const refreshQuiz = useCallback((payload: ServerQuizRequestPayload) => {
        console.log("refreshQuiz");
        setQuiz(payload.data.questions);
        setQuizID(payload.data.quiz_id);
        fillcheckboxValues(payload.data); //TODO update fillcheckboxValues implementation to accept quiz type in parameters
    }, []);
```